slack_notifier.rb

```diff
@@ -0,0 +1,42 @@
+require "json"
+require "curb"
+
+class SlackNotifier
+  def initialize(slack_webhook)
+    @slack_webhook = slack_webhook
+  end
+
+  def new_issue_not_published
+    send_message("*Fíha, dnes na ÚVO nevyšlo nové vydanie vestníka?*")
+  end
+
+  def matching_announcements(page, announcements)
+    send_header_message(page, announcements)
+
+    announcements.each do |a|
+      send_message("<#{a[:link][:href]}|#{a[:link][:text]}>: *#{a[:customer]}* #{a[:description]}")
```

Comment:
Zhrniem to:

- WebMock - ja som za každú simplifikáciu, skúsim to prerobiť s Curl doublom.
- Timecop je tam kvôli tomu aby sedel dátum fixtúry s Date.today, toho by sa tiež dalo zbaviť keby tá metóda berie dátum ako parameter.
- Z názvu metódy som naschvál vynechal "found", lebo ona podľa implementácie môže aj nenájsť, vtedy je 0 oznamov. A found mi znelo ako úspešné dokonanie. Celkovo tá metóda robí dve veci a skôr by som ju mal rozdeliť  na matching_announcements_found a na no_announcements_found.